### PR TITLE
Refactoring monitor external transactions

### PR DESCRIPTION
**Pull request by Griffics**

### CODE REVIEW

File: Context/Application/Cli/CliModule.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+            useClass: TypeOrmRepository
+        },
+        {
+            provide: "DepositRequestRepositoryInterface",
+            useClass: DepositRequestTypeOrmRepository
+        }
+
+    ]
+})
+export class CliModule {}
```
Comment (walteratwork) [inline Context/Application/Cli/CliModule.ts]: No newline at end of file. Check all new files for this issue.

File: Context/Application/Command/ExternalBlockchain/ProcessIncomingContractCreation/ProcessIncomingContractCreation.ts [hunk 2]
```diff
@@ -1,4 +1,7 @@
-export default class ProcessIncomingContractCreation {
+import ExternalBlockchainCommandInterface
+    from "context/Application/Command/ExternalBlockchain/ChainProcessor/ExternalBlockchainCommandInterface";
+
+export default class ProcessIncomingContractCreation implements ExternalBlockchainCommandInterface{
```
Comment (walteratwork) [inline Context/Application/Command/ExternalBlockchain/ProcessIncomingContractCreation/ProcessIncomingContractCreation.ts]: Missing space before curly brace. Could we use eslint to handle such issues?

File: Context/Application/Command/ExternalBlockchain/ChainProcessor/ExternalBlockchainCommandInterface.ts [hunk 3]
```diff
@@ -0,0 +1,4 @@
+export default interface ExternalBlockchainCommandInterface {
+    get txHash(): string
+    get contractId(): string
+}
```
Comment (walteratwork) [inline Context/Application/Command/ExternalBlockchain/ChainProcessor/ExternalBlockchainCommandInterface.ts]: Why do you assume that all commands will have this data?

File: Context/Application/Command/ExternalBlockchain/ChainProcessor/ExternalBlockchainHandlerInterface.ts [hunk 4]
```diff
@@ -0,0 +1,6 @@
+import BlockRange from "context/Application/Command/ExternalBlockchain/ChainProcessor/BlockRange";
+
+export default interface ExternalBlockchainHandlerInterface {
+    execute(range: BlockRange): Promise<void>
```
Comment (walteratwork) [inline Context/Application/Command/ExternalBlockchain/ChainProcessor/ExternalBlockchainHandlerInterface.ts]: Why do you assume that all command handlers will have an execute method with a range parameter?

File: Context/Application/Command/ExternalBlockchain/ChainProcessor/ExternalBlockchainHandlerInterface.ts [hunk 5]
```diff
@@ -0,0 +1,6 @@
...
+
```
Comment (walteratwork) [inline Context/Application/Command/ExternalBlockchain/ChainProcessor/ExternalBlockchainHandlerInterface.ts]: excess line

File: Context/Application/Command/ExternalBlockchain/ProcessIncomingContractCreation/ProcessIncomingContractCreationHandler.ts [hunk 6]
```diff
@@ -13,40 +11,60 @@ import Deposit from "context/Domain/Deposit";
...
-        private externalBlockchain: ExternalBlockchain
+        @Inject("DepositRepositoryInterface") private repository: DepositRepositoryInterface,
+        @Inject("DepositRequestRepositoryInterface") private depositRequestRepository: DepositRequestRepositoryInterface,
+        private externalBlockchain: ExternalBlockchain,
+        private getLastContractsHandler: GetLastContractsHandler
     ) {}
 
-    async execute(command: ProcessIncomingContractCreation): Promise<void> {
-        const txIncluded = await this.externalBlockchain.repository.txIncluded(command.txHash)
+    async execute(range: BlockRange): Promise<void> {
```
Comment (walteratwork) [inline Context/Application/Command/ExternalBlockchain/ProcessIncomingContractCreation/ProcessIncomingContractCreationHandler.ts]: I prefer the previous version where the command handler receives the command to handle.
This follows the Command pattern.

**Review by walteratwork — CHANGES_REQUESTED** [review]
Fix merge conflicts and make tests work

File: Context/Application/Command/ExternalBlockchain/ProcessIncomingContractCreation/ProcessIncomingContractCreation.ts (hunk 2 above)
Comment (lunacrafts) [reply Context/Application/Command/ExternalBlockchain/ProcessIncomingContractCreation/ProcessIncomingContractCreation.ts]: [editorconfig](https://editorconfig.org/) should work great in this and above case

Comment (Griffics) [reply Context/Application/Command/ExternalBlockchain/ProcessIncomingContractCreation/ProcessIncomingContractCreation.ts]: > [editorconfig](https://editorconfig.org/) should work great in this and above case

Thank you for interesting idea

**Review by walteratwork — APPROVED** [review]

**Pull request merged**